File: meWrap/Models/ImageEditor/ImageEditor.m
//
//  ImageEditor.m
//  meWrap
//
//  Created by Sergey Maximenko on 10/20/15.
//  Copyright © 2015 Ravenpod. All rights reserved.
//

#import "ImageEditor.h"
#import <AdobeCreativeSDKCore/AdobeCreativeSDKCore.h>
#import <AdobeCreativeSDKImage/AdobeCreativeSDKImage.h>

@interface ImageEditor ()

@end

@implementation ImageEditor

+ (void)editImage:(UIImage *)image completion:(ImageBlock)completion cancel:(Block)cancel {
    UIViewController *presentingViewController = [UIWindow mainWindow].rootViewController;
    AdobeUXImageEditorViewController* controller = (id)[self editControllerWithImage:image completion:^(UIImage *image) {
        if (completion) completion(image);
        [presentingViewController dismissViewControllerAnimated:NO completion:nil];
    } cancel:^ {
        if (cancel) cancel();
        [presentingViewController dismissViewControllerAnimated:NO completion:nil];
    }];
    
    [presentingViewController presentViewController:controller animated:NO completion:nil];
}

+ (AdobeUXImageEditorViewController*)editControllerWithImage:(UIImage*)image completion:(ImageBlock)completion cancel:(Block)cancel {
    static dispatch_once_t onceToken;
    dispatch_once(&onceToken, ^{
        [AdobeImageEditorCustomization setSupportedIpadOrientations:@[@(UIInterfaceOrientationPortrait),@(UIInterfaceOrientationPortraitUpsideDown),@(UIInterfaceOrientationLandscapeLeft),@(UIInterfaceOrientationLandscapeRight)]];
        [[AdobeUXAuthManager sharedManager] setAuthenticationParametersWithClientID:@"a7929bf566694d579acb507eae697db1"
                                                                       clientSecret:@"b6fa1e1c-4f8c-4001-88a9-0251a099f890" enableSignUp:NO];
    });
    AdobeUXImageEditorViewController* controller = [[AdobeUXImageEditorViewController alloc] initWithImage:image];
    [controller enqueueHighResolutionRenderWithImage:image completion:^(UIImage *result, NSError *error) {
        if (result) {
            if (completion) completion(result);
        } else {
            if (cancel) cancel();
        }
    }];
    return controller;
}

@end

File: meWrap/Models/ImageEditor.swift
//
//  ImageEditor.swift
//  meWrap
//
//  Created by Sergey Maximenko on 2/25/16.
//  Copyright © 2016 Ravenpod. All rights reserved.
//

import Foundation

class ImageEditor {
    
    class func editImage(image: UIImage, completion: UIImage -> Void) {
        let presentingViewController = UINavigationController.main
        let controller = editControllerWithImage(image, completion: { image in
            completion(image)
            presentingViewController.dismissViewControllerAnimated(false, completion: nil)
            }, cancel: {
                presentingViewController.dismissViewControllerAnimated(false, completion: nil)
            })
        presentingViewController.presentViewController(controller, animated: false, completion: nil)
    }
    
    private static var token: dispatch_once_t = 0
    
    class func editControllerWithImage(image: UIImage, completion: UIImage -> Void, cancel: Block) -> AdobeUXImageEditorViewController {
        
        dispatch_once(&token) {
            AdobeImageEditorCustomization.setSupportedIpadOrientations([UIInterfaceOrientation.Portrait.rawValue, UIInterfaceOrientation.PortraitUpsideDown.rawValue, UIInterfaceOrientation.LandscapeLeft.rawValue, UIInterfaceOrientation.PortraitUpsideDown.rawValue, UIInterfaceOrientation.LandscapeRight.rawValue])
            AdobeImageEditorCustomization.setToolOrder([
                kAdobeImageEditorEnhance,
                kAdobeImageEditorEffects,
                kAdobeImageEditorStickers,
                kAdobeImageEditorOrientation,
                kAdobeImageEditorCrop,
                kAdobeImageEditorColorAdjust,
                kAdobeImageEditorLightingAdjust,
                kAdobeImageEditorSharpness,
                kAdobeImageEditorDraw,
                kAdobeImageEditorText,
                kAdobeImageEditorRedeye,
                kAdobeImageEditorWhiten,
                kAdobeImageEditorBlemish,
                kAdobeImageEditorBlur,
                kAdobeImageEditorMeme,
                kAdobeImageEditorFrames,
                kAdobeImageEditorFocus,
                kAdobeImageEditorSplash,
                kAdobeImageEditorVignette
                ])
            AdobeUXAuthManager.sharedManager().setAuthenticationParametersWithClientID("a7929bf566694d579acb507eae697db1", withClientSecret: "b6fa1e1c-4f8c-4001-88a9-0251a099f890")
        }
        let controller = AdobeUXImageEditorViewController(image:image)
        controller.enqueueHighResolutionRenderWithImage(image) { (result, error) -> Void in
            if let result = result {
                completion(result)
            } else {
                cancel()
            }
        }
        return controller
    }
    
}